stdlib/public/runtime/HeapObject.cpp

```diff
@@ -849,6 +849,15 @@ void swift::swift_deallocPartialClassInstance(HeapObject *object,
   // Destroy ivars
   auto *classMetadata = _swift_getClassOfAllocated(object)->getClassObject();
   assert(classMetadata && "Not a class?");
+
+  // If the object's class is already pure ObjC class, just release it and move
+  // on. There are no ivar destroyers. This avoids attempting to mutate
+  // placeholder objects statically created in read-only memory.
+  if (classMetadata->isPureObjC()) {
+    objc_release((id)object);
```

Comment:
Hmmm. Looks like this file somehow doesn't include the definition of `id`; seems odd that it's got `objc_release()`, but not `id`.

Reply:
Oh silly me, this needs to be guarded by `OBJC_INTEROP` too.